devicetypes/fibargroup/fibaro-door-window-sensor-2.src/fibaro-door-window-sensor-2.groovy

```diff
@@ -129,10 +131,8 @@ def updated() {
 	}
 	logging("${device.displayName} - Executing updated()","debug")
 
-	if ( settings.temperatureHigh as Integer == 0 && settings.temperatureLow as Integer == 0 ) { 
-		sendEvent(name: "temperatureAlarm", value: null, displayed: false) 
-	} else if ( settings.temperatureHigh != null || settings.temperatureHigh != null ) {
-		sendEvent(name: "temperatureAlarm", value: "clear", displayed: false) 
+	if (settings.temperatureHigh != null || settings.temperatureLow != null) {
+        sendEvent(name: "temperatureAlarm", value: "cleared", displayed: false) 

```

Comment:
I think the other case should probably result in us sending a "cleared" as well, although I can't say I understand what the original intent was

Reply:
It will anyway send "cleared" event now so I wanted to optimize that with only one line.

Reply:
I don't understand. How are we still sending a cleared event if `temperatureHigh` and `temperatureLow` are both `0` now?

Reply:
@greens if the intention is to sendEvent(name: "temperatureAlarm", value: "cleared", displayed: false) in both cases then this condition should be enough
`if (settings.temperatureHigh != null || settings.temperatureLow != null) {
        sendEvent(name: "temperatureAlarm", value: "cleared", displayed: false)
}`

Reply:
oh, you're totally right.

duh